### PR TITLE
Initial stack states implementation

### PR DESCRIPTION
**Pull request by bquenin**
Related to #162 
## Verification

```
$ sudo ./swarm init
$ make test-stack
```

### CODE REVIEW

File: api/rpc/stack/stack-state.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
+package stack
+
+import (
+	"fmt"
+	"sync"
+)
+
+// State holds the current stack state, and has methods to get and
```
Comment (generalhenry) [inline api/rpc/stack/stack-state.go]: Shouldn't the state be stored in etcd?

Comment (bquenin) [reply api/rpc/stack/stack-state.go]: It definitely should! :)

**Review by generalhenry — APPROVED** [review]

File: api/rpc/stack/stack-state.go (hunk 1 above)
Comment (subfuzion) [reply api/rpc/stack/stack-state.go]: hehe, we already had that conversation out of band.

**Review by subfuzion — COMMENTED** [review]
Per our earlier conversation, goals include
- [x] convert to a state transition table
- [x] remove lock: we'll synchronize on etcd
- [ ] always query etcd state for current state before validating/setting new state transition
- [ ] provide ability to supply the transition table

**Pull request closed without merging**